### PR TITLE
Add YAML autodocs to website documentation

### PR DESCRIPTION
**Pull request by tempoz**
Have to create `.mdx` versions of the YAML files so we can transclude them in the markdown.

---

**Version bump**: Patch

### CODE REVIEW

**Comment by tempoz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2183** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2183" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/2183?utm_source=stack-comment).

**Comment by tempoz** [comment]
This does make the `website` targets depend on code in the `enterprise` directory, not sure if that's a concern.

File: docs/config-options-listings.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+names of the nesting structures, if any, and the option name with a `.`
+separating each element.
+
+E.g.,
+```yaml
+storage:
+    disk:
+        root_directory: /tmp/buildbuddy
+```
+becomes:
```
Comment (bduffany) [inline docs/config-options-listings.mdx:23]: nit: show a command-line example for `StructSliceFlag` too (since the JSON syntax is a bit trickier?)

File: docs/config-options-listings.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+```sh
+buildbuddy -storage.disk.root_directory="/tmp/buildbuddy"
+```
+
+### Buildbuddy Server (Free Tier)
+
+<FreeServerConfig />
```
Comment (bduffany) [inline docs/config-options-listings.mdx:30]: not specific to this PR, but some thoughts on readability of the generated options

```yaml
    tree_cache_seed: treecache # If set, hash this with digests before caching / reading from tree cache type: string
```

this is a bit hard to read because it requires horizontal scrolling, and also it's a bit hard to parse the "type" part because it doesn't end with a period.

what do you think about adding some primitive line-wrapping and also putting the type in parenthesis to more clearly separate it from the help content?

```yaml
    # The desired log level. Logs with a level >= this level will be emitted.
    # One of {'fatal', 'error', 'warn', 'info', 'debug'} (type: string)
    log_level: info
```

File: docs/config-options-listings.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import EnterpriseServerConfig from '../bazel-bin/enterprise/server/cmd/server/yaml_doc/buildbuddy_enterprise_server_documented_defaults.mdx'
+import ExecutorConfig from '../bazel-bin/enterprise/server/cmd/executor/yaml_doc/buildbuddy_executor_documented_defaults.mdx'
+
+Provided below are working, documented YAML configs for each Buildbuddy binary
+containing every option that that binary accepts, each set to the default value
+for that option. Any option can be specified on the command line by joining the
+names of the nesting structures, if any, and the option name with a `.`
+separating each element.
+
+E.g.,
```
Comment (bduffany) [inline docs/config-options-listings.mdx:17]: nit: I remember reading a technical writing "best practice" somewhere not to use "E.g." since it's not as easily recognized by non-native speakers. It recommended "For example:" as an alternative.

File: docs/config-options-listings.mdx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+storage:
+    disk:
+        root_directory: /tmp/buildbuddy
+```
+becomes:
+```sh
+buildbuddy -storage.disk.root_directory="/tmp/buildbuddy"
+```
+
+### Buildbuddy Server (Free Tier)
```
Comment (bduffany) [inline docs/config-options-listings.mdx:28]: I don't know whether we usually document this as the "free tier" server or "OSS server" or something else, @siggisim might have thoughts

File: docs/config-options-listings.mdx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+sidebar_label: Options Listings
+---
+
+import FreeServerConfig from '../bazel-bin/server/cmd/buildbuddy/yaml_doc/buildbuddy_server_documented_defaults.mdx'
+import EnterpriseServerConfig from '../bazel-bin/enterprise/server/cmd/server/yaml_doc/buildbuddy_enterprise_server_documented_defaults.mdx'
+import ExecutorConfig from '../bazel-bin/enterprise/server/cmd/executor/yaml_doc/buildbuddy_executor_documented_defaults.mdx'
+
+Provided below are working, documented YAML configs for each Buildbuddy binary
+containing every option that that binary accepts, each set to the default value
+for that option. Any option can be specified on the command line by joining the
```
Comment (bduffany) [inline docs/config-options-listings.mdx:13]: nit: I found the second sentence a bit hard to parse -- maybe split it up and reword slightly like:

Any options that can be specified in the YAML config can also be passed at the command line. For nested options, be sure to write out the full YAML path, with a `.` separating each part.

File: docs/config-options-listings.mdx (hunk 2 above)
Comment (bduffany) [reply docs/config-options-listings.mdx:30]: (or maybe for convenience, you could also list out the full YAML path like:)

```yaml
  # cache.tree_cache_seed (string): If set, hash this with digests before
  # caching / reading from tree cache
  tree_cache_seed: treecache
```

File: docs/config-options-listings.mdx (hunk 4 above)
Comment (siggisim) [reply docs/config-options-listings.mdx:28]: Yeah, something like ### BuildBuddy Server (Open Source) would align with the helm charts: https://github.com/buildbuddy-io/buildbuddy-helm/tree/master/charts/buildbuddy

File: docs/config-options-listings.mdx (hunk 3 above)
Comment (tempoz) [reply docs/config-options-listings.mdx:17]:
```suggestion
For example:
```

File: docs/config-options-listings.mdx (hunk 2 above)
Comment (bduffany) [reply docs/config-options-listings.mdx:30]: sgtm!

File: docs/config-options-listings.mdx (hunk 1 above)
Comment (tempoz) [reply docs/config-options-listings.mdx:23]: Done!

File: docs/config-options-listings.mdx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+```yaml
+cache:
+    disk:
+        partitions: [] # type: []disk.Partition
+        - id: "1GB"
+          max_size_bytes: 1073741824
+        - id: "2GB":
+          max_size_bytes: 2147483648
+```
+becomes:
```
Comment (bduffany) [inline docs/config-options-listings.mdx:42]: nit: it could be a bit more clear on how to determine whether something is specified as a struct or not. maybe something like:

---

For example, given the following schema:
```yaml
cache:
    disk:
        partitions: [] # type: []disk.Partition
        # e.g.:
        # - id: "" # type: string
        #   max_size_bytes: 0 # type: int
```

We see that `cache.disk.partitions` is configured as a list of `disk.Partition`. In YAML, we'd normally configure it like this:

```yaml
cache:
    disk:
        partitions:
            - id: "1GB"
              max_size_bytes: 1073741824
            - id: "2GB":
              max_size_bytes: 2147483648
```

The flag equivalent of this example would be:
...

**Review by bduffany — APPROVED** [review]
LGTM, just 1 nit

**Review by bduffany — APPROVED** [review]

File: docs/config-options-listings.mdx (hunk 6 above)
Comment (tempoz) [reply docs/config-options-listings.mdx]: Done!

**Review by bduffany — APPROVED** [review]

File: docs/config-options-listings.mdx [hunk 7]
```diff
@@ -0,0 +1,70 @@
+---
+id: config-options-listings
+title: Listings For All BuildBuddy Options
+sidebar_label: Options Listings
```
Comment (siggisim) [inline docs/config-options-listings.mdx:4]: Maybe we should call this "All Options" for the id, title, and sidebar label (you'll need to change it in sidebars.js too)

**Review by siggisim — APPROVED** [review]
LGTM, just one nit

File: docs/config-options-listings.mdx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        partitions: [] # type: []disk.Partition
+        # e.g.:
+        # - id: "" # type: string
+        #   max_size_bytes: 0 # type: int
+```
+We see that `cache.disk.partitions` is configured as a list of `disk.Partition`. In YAML, we'd normally configure it like this:
+```yaml
+cache:
+    disk:
+        partitions: [] # type: []disk.Partition
```
Comment (bduffany) [inline docs/config-options-listings.mdx:45]:
```suggestion
        partitions:
```

Comment (tempoz) [reply docs/config-options-listings.mdx:45]: Addressed in #2206
